### PR TITLE
Add function `use_rscloud_badge()`

### LINKED ISSUE #1584: Add function `use_rscloud_badge()`
It would be great if usethis could provide a helper function that creates a README badge for existing RStudio Cloud projects.

The proposed `use_rscloud_badge()` function would create a README badge that could link to either public or private RStudio Cloud projects, since the only parameter required would be the link to the existing RStudio Cloud project. Such a README badge would be immediately useful to those who teach classes or share research compendia with RStudio Cloud. Another value added is that this function could standardize what RStudio Cloud README badges look like across GitHub.

I created a draft badge URL for your consideration using shields.io, shown below. It uses the color shown [here](https://simpleicons.org/?q=rstudio) and the RStudio logo which was already accessible on shields.io from the list of "Simple Icon slugs" [here](https://github.com/simple-icons/simple-icons/blob/develop/slugs.md):

Badge: launch cloud
URL: https://img.shields.io/badge/launch-cloud-75AADB?style=flat&logo=rstudio

The use case is not easy to show in a reprex since it requires knitting an .Rmd, so instead I've created a test GitHub repository [here](https://github.com/gvelasq/rscloud-test) and used it to spin up a public RStudio Cloud project [here](https://rstudio.cloud/project/3584129). The functionality that I'm proposing for `use_rscloud_badge("https://rstudio.cloud/project/3584129")` would be equivalent to calling:

```r
use_badge("Launch RStudio Cloud", "https://rstudio.cloud/project/3584129", "https://img.shields.io/badge/launch-cloud-75AADB?style=flat&logo=rstudio")
```

The implementation of `use_rscloud_badge()` would be along these lines in [badge.R](https://github.com/r-lib/usethis/blob/HEAD/R/badge.R):

```r
#' @rdname badges
#' @param cloudurl A link to an existing RStudio Cloud project
#' @export
use_rscloud_badge <- function(cloudurl) {
  img <- "https://img.shields.io/badge/launch-cloud-75AADB?style=flat&logo=rstudio"
  use_badge("Launch RStudio Cloud", cloudurl, img)

  invisible(TRUE)
}
```

### PR DESCRIPTION
**Pull request by gvelasq**
Closes #1584

Added a new function `use_rscloud_badge()` that creates a README badge indicating the repository can be launched in an [RStudio Cloud](https://rstudio.cloud) project.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # usethis (development version)
 
+`use_rscloud_badge()` is a new function that creates a README badge indicating the repository can be launched in an [RStudio Cloud](https://rstudio.cloud) project (@gvelasq, #1589).
```
Comment (jennybc) [inline NEWS.md:3]: We usually prefer to list the issue number. Or the issue number and the PR number.

File: R/badge.R [hunk 2]
```diff
@@ -134,6 +135,22 @@ use_binder_badge <- function(ref = git_default_branch(), urlpath = NULL) {
   invisible(TRUE)
 }
 
+#' @rdname badges
+#' @param cloudurl A link to an existing [RStudio Cloud](https://rstudio.cloud) project
```
Comment (jennybc) [inline R/badge.R:139]: Do we need to say anything more about how to get this URL? Or link to any Cloud docs? Pardon my ignorance, I just haven't ever sent a bunch of people to an RStudio Cloud project like this.

File: R/badge.R [hunk 3]
```diff
@@ -134,6 +135,22 @@ use_binder_badge <- function(ref = git_default_branch(), urlpath = NULL) {
...
+#' @export
+use_rscloud_badge <- function(cloudurl) {
```
Comment (jennybc) [inline R/badge.R:141]: Maybe `url` is a better name?

File: R/badge.R [hunk 4]
```diff
@@ -134,6 +135,22 @@ use_binder_badge <- function(ref = git_default_branch(), urlpath = NULL) {
...
+  if (grepl("rstudio.cloud/project", cloudurl)) {
+    img <- "https://img.shields.io/badge/launch-cloud-75aadb?style=flat&logo=rstudio"
+    use_badge("Launch RStudio Cloud", cloudurl, img)
+  } else {
+    ui_stop("
+    {ui_code('usethis::use_rscloud_badge()')} requires a link to an existing RStudio Cloud project of the form 'https://rstudio.cloud/project/<id>'.
+    ")
```
Comment (jennybc) [inline R/badge.R:148]: Can you look around for some other multi-line `ui_stop()`s and handle line breaking and whitespace as they do?

File: NEWS.md (hunk 1 above)
Comment (gvelasq) [reply NEWS.md]: Thank you for clarifying. Fixed in b30dc78bfc4d86c034057ec4f192eff25e5fe947.

File: R/badge.R (hunk 4 above)
Comment (gvelasq) [reply R/badge.R]: Apologies, I used the style currently in use for `ui_oops()` in [`use_badge()`](https://github.com/r-lib/usethis/blame/main/R/badge.R#L43-L47).

In 2abc138c6944cac769edfbf2061e4e4ffbc3edb7, I changed `ui_stop()` to use the style currently used in [`use_github_release()`](https://github.com/r-lib/usethis/blame/main/R/release.R#L177-L179).

**Comment by gvelasq** [comment]
I recently learned that another valid path for RStudio Cloud projects, when using shared spaces, is of the form
```html
https://rstudio.cloud/spaces/<space-id>/project/<project-id>
```
I added logic to support this type of URL and an additional test for `use_rscloud_badge()` in cd6f856be8c23452970a0d5d7670e684dc16a705.

**Comment by jennybc** [comment]
Thanks!

**Pull request merged**